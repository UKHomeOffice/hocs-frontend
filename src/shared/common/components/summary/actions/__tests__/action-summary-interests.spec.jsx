import React from 'react';
import ActionSummaryInterests from '../action-summary-interests';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';

describe('action-summary-interests.jsx', () => {

    it('should render one with details and one without', () => {

        const props = {
            items: [
                {
                    interestedPartyTitle: 'Interested party 1',
                    detailsOfInterest: 'Details of interest',
                },{
                    interestedPartyTitle: 'Interested party 1',
                },
            ]
        };

        expect(render(
            <>
                <ActionSummaryInterests items={props.items} />
            </>
        )).toMatchSnapshot();
    });
});
