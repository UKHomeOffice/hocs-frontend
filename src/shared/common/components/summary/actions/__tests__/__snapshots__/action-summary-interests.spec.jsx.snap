// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`action-summary-interests.jsx should render one with details and one without 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h2
        class="govuk-heading-m"
      >
        Interest
      </h2>
      <table
        class="govuk-table margin-left--small"
      >
        <caption
          class="govuk-table__caption margin-bottom--small"
        >
          Interest
        </caption>
        <tbody
          class="govuk_table__body"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Party
            </th>
            <td
              class="govuk-table__cell"
            >
              Interested party 1
            </td>
          </tr>
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Details
            </th>
            <td
              class="govuk-table__cell"
            >
              Details of interest
            </td>
          </tr>
        </tbody>
      </table>
      <table
        class="govuk-table margin-left--small"
      >
        <caption
          class="govuk-table__caption margin-bottom--small"
        >
          Interest
        </caption>
        <tbody
          class="govuk_table__body"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header padding-left--small govuk-!-width-one-third"
            >
              Party
            </th>
            <td
              class="govuk-table__cell"
            >
              Interested party 1
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </body>,
  "container": <div>
    <h2
      class="govuk-heading-m"
    >
      Interest
    </h2>
    <table
      class="govuk-table margin-left--small"
    >
      <caption
        class="govuk-table__caption margin-bottom--small"
      >
        Interest
      </caption>
      <tbody
        class="govuk_table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Party
          </th>
          <td
            class="govuk-table__cell"
          >
            Interested party 1
          </td>
        </tr>
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Details
          </th>
          <td
            class="govuk-table__cell"
          >
            Details of interest
          </td>
        </tr>
      </tbody>
    </table>
    <table
      class="govuk-table margin-left--small"
    >
      <caption
        class="govuk-table__caption margin-bottom--small"
      >
        Interest
      </caption>
      <tbody
        class="govuk_table__body"
      >
        <tr
          class="govuk-table__row"
        >
          <th
            class="govuk-table__header padding-left--small govuk-!-width-one-third"
          >
            Party
          </th>
          <td
            class="govuk-table__cell"
          >
            Interested party 1
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
